File: api/src/main/java/dev/obscuria/fragmentum/registry/DeferredAttribute.java
package dev.obscuria.fragmentum.registry;

import net.minecraft.core.Holder;
import net.minecraft.world.entity.ai.attributes.Attribute;

import java.util.function.Supplier;

@SuppressWarnings("unused")
public class DeferredAttribute extends Deferred<Attribute, Attribute>
{
    public DeferredAttribute(Supplier<Holder<Attribute>> supplier)
    {
        super(supplier);
    }
}

File: api/src/main/kotlin/dev/obscuria/fragmentum/registry/DeferredAttribute.kt
@file:Suppress("unused", "RedundantNullableReturnType")

package dev.obscuria.fragmentum.registry

import net.minecraft.core.Holder
import net.minecraft.world.entity.ai.attributes.Attribute

class DeferredAttribute(supplier: () -> Holder<Attribute>) : Deferred<Attribute, Attribute>(supplier)